src/jgikbase/idmapping/core/namespaced_id.py

```diff
@@ -0,0 +1,17 @@
+"""
```

Comment:
having one module called `namespaced_id.py` and one module called `namespace_id.py` seems very confusing and a bit error prone. I definitely can't tell any semantic meaning or difference for entities in the system based on those names

Also, the name "Namespace ID" does not convey to me any meaning. To me, "namespace" and "namespace ID" refer 
 to the same thing. Since this is supposed to be a mapping from a namespace to an object ID, can you instead call it "namespace_mapping"? I find that more descriptive.

Reply:
I've thought the same thing but haven't yet come up with a name I like. Some I've considered are
`id_in_namespace`
`namespace_and_id`
`id_with_namespace`

Basically one is a namespace that contains IDs, and one is an ID in a particular namespace.

> Also, the name "Namespace ID" does not convey to me any meaning. To me, "namespace" and "namespace ID" refer to the same thing.

A namespace can have many properties, one of which is the id of the namespace.

Reply:
Per conversation:
namespaced_id -> object_id, which includes a namespace ID since all object ids are associated with a namespace

Reply:
I'll do that in a future PR if that's ok.